### ISSUE TITLE
Adam seems to be broken

### ISSUE THREAD

**Issue by RanjithGHegde** [OpenNMT/OpenNMT-py#805]
When using adam optimizer the ppl stays a high constant value, with accuracy varying erratically.

**Comment by RanjithGHegde** [comment]
![image](https://user-images.githubusercontent.com/39875415/42412234-e0dcf22c-8225-11e8-8268-c0bf65b24048.png)

python train.py -data data/demo -save_model demo-model -optim adam -gpuid 0
that is the command i am using and yes i am on master.

This is the issue:
![image](https://user-images.githubusercontent.com/39875415/42412240-fd4efebe-8225-11e8-9b49-8a3d708b1955.png)
 I am using default dataset itself.

**Comment by vince62s** [comment]
you need to set the learning_rate to 0.001
by default it's 1 because default optim is sgd

**Comment by RanjithGHegde** [comment]
Thanks I had overlooked that one 😅
